rtmp-client/src/main/java/net/butterflytv/rtmp_client/RtmpClient.java

```diff
@@ -13,8 +13,18 @@
 
 
     private final static int OPEN_SUCCESS = 1;
+    private final static int TIMEOUT = 10;
     private long rtmpPointer = 0;
 
+    /* Timeout value in seconds */
```

Comment:
It seems strange that this is in whole seconds. Normally I'd expect to specify a timeout in milliseconds. Is this possible?

Reply:
We initially considered setting the timeout in milliseconds. The only conflict we saw was that there is already a timeout in the rtmp library but it was being used to set only the socket's receive timeout. So we favoured using the same.
Do you think it would be better to use a different value for the send timeout (in ms), which can be set from the application and continue using the earlier timeout for receive?

Reply:
> The only conflict we saw was that there is already a timeout in the rtmp library but it was being used to set only the socket's receive timeout. So we favoured using the same.

I'm confused about how this is relevant to the unit being used. Please clarify. Is it that seconds is used somewhere elsewhere (if so, where abouts)?

Reply:
There is a rtmp->Link.timeout variable which holds the timeout in seconds. It is being used here - https://github.com/ittiam-systems/LibRtmp-Client-for-Android/blob/bug-67/rtmp-client/src/main/cpp/librtmp/rtmp.c#L950.
I could add another timeoutInMs variable to the rtmp structure and keep the receive timeout and send timeout different.

Reply:
* If it's possible to set them separately, then I think they should be separate and both settable via the Java API
* All Java timeouts should be in the same unit. Ideally milliseconds, since this is more standard for a Java API. However if there's already a Java API that takes seconds that we need to remain consistent with (I couldn't see one?), or if a value passed is going to be rounded to seconds internally (i.e. a millisecond value would lose precision internally) then it's indeed better to use seconds. Either way, let's make sure the unit is well documented on the Java APIs.

Reply:
There are two APIs now, one to set the send timeout (in milliseconds) and another to set the receive timeout (in seconds).
The javadoc mention the units for both the timeouts and even the variable names used indicate the units.

I have kept the receive timeout unit in seconds since the variable that is used, rtmp->Link.timeout is also set using another C API called [RTMP_SetupStream()](https://github.com/ittiam-systems/LibRtmp-Client-for-Android/blob/bug-67/rtmp-client/src/main/cpp/librtmp/rtmp.c#L431).

Reply:
> I have kept the receive timeout unit in seconds since the variable that is used, rtmp->Link.timeout is also set using another C API called RTMP_SetupStream().

Can we not just change the implementation of `RTMP_SetupStream()`, and then use milliseconds? I would really rather just have everything specified in milliseconds at the Java layer, if at all possible. Mixing units for timeouts in an API seems quite non-ideal.

Reply:
Done. Changed to milliseconds everywhere.